### ISSUE TITLE
In-Game: [YYC] Silent crash when a function reads a variable from chained custom constructors

### ISSUE THREAD

**Issue by SilentPhil** [YoYoGames/GameMaker-Bugs#7453]
### Description

Calling two custom user functions in a certain way results in a crash:

```
**********************************.
Entering main loop.
**********************************.
Pause event has been registered for this frame
Pause event has been unregistered
func: Foo
func: Foo
0
0
func: Foo


D:\GameMakerCacheSteam\GMS2TEMP\SilentCrashTest2024_29E2245D_YYC\SilentCrashTest2024.exe exited with non-zero status (-1073741819)
elapsed time 00:00:43.5840475s for command "C:\ProgramData\GameMakerStudio2\Cache\runtimes\runtime-2024.6.1.208/bin/igor/windows/x64/Igor.exe" -j=16  -options="D:\GameMakerCacheSteam\GMS2TEMP\build.bff" -v -- Windows Run started at 08/28/2024 13:06:49
FAILED: Run Program Complete
```

Reproduced only when assembled with YYC.

In the attached project you can observe the explicit scenario for reproduction.

[contact_us.zip](https://github.com/user-attachments/files/16781074/contact_us.zip)

### Expected Change

The game shouldn't crash

### Steps To Reproduce

1. Start GameMaker
2. Open the attached project
3. Press W or R 
4. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.6.2 (Monthly), 2024.800 (Betas)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

<img width="1024" alt="windbg" src="https://github.com/user-attachments/assets/c5a77010-b59b-489b-931b-908a85029583">

**Comment by YYDan** [comment]
Certainly, your sample reproduces the problem for me in IDE v2024.800.0.614  Runtime v2024.800.0.637, so all good there - ta.

**Comment by rwkay** [comment]
Fixed in 2024.10

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.1100.0.626  Runtime v2024.1100.0.652
